app/services/stripe_subscription_service.rb

```
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+# Object in charge of users subscriptions on the Stripe platform
+# stripe service: billing
+class StripeSubscriptionService
+  PLANS = {
+    premium: 'plan_G5Sw4jL2FPMk5m',
+    enterprise: 'plan_G5SsKKgPBo3lvU'
+  }.freeze
+
+  attr_reader :account
+
+  def initialize(account)
+    @account = account
+  end
+
+  def subscribe_to(subscription_type)
+    plan = find_product(subscription_type)
+    if plan
+      subscription = Stripe::Subscription.create(
+        customer: account.customer_id,
+        items: [{ plan: plan }]
+      )
+      account.update_attributes(subscription_id: subscription.id)
+      return subscription
+    else
+      false
```

Maybe we can throw a nice expception here, instead of returning false?

---

Sounds legit. I basically want to render an error but not sure if I can call render from here without a double render error 🤔

---

Which is why I am returning false and then rendering an error in the controller.